settings.py

```diff
@@ -808,6 +808,10 @@
 S3_ACCESS_KEY = None
 S3_SECRET_KEY = None
 
+# Takes no arguments and returns a valid bucket naming scheme
+# ex: BUCKET_NAME_FUNCTION = lambda: datetime.datetime.utcnow().strftime('%Y-%m-%d')
+BUCKET_NAME_FUNCTION = None
```

Comment:
I think this should be a function path string rather than a function itself. In the actual implementation it might even make sense for this to be a callable class so that it can maintain some state about which buckets have been created and also create the new buckets when necessary.